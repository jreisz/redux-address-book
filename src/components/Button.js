import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ onClick, className, text }) => (
  <button onClick={onClick} className={className} type="button">
    {text}
  </button>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  text: PropTypes.string,
};

Button.defaultProps = {
  className: 'a-class-name',
  text: 'Button text',
};

export default Button;
