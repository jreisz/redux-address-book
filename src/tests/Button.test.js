import React from 'react';
import { shallow } from 'enzyme';
import './enzymeAdapter';
import Button from '../components/Button';

const mockOnClick = jest.fn();
const mockProps = {
  onClick: mockOnClick,
  className: 'a-class-name',
  text: 'Click me!',
};

describe('Button', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Button {...mockProps} />);
  });

  it('renders like expected (snapshot)', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
