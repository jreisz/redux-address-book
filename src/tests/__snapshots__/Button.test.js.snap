// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders like expected (snapshot) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className="a-class-name"
    onClick={[Function]}
    text="Click me!"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Click me!",
      "className": "a-class-name",
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": "Click me!",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Click me!",
        "className": "a-class-name",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": "Click me!",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
